webhooks/pod_webhook.go

```diff
@@ -149,75 +112,283 @@ func (m *PodMutator) Handle(ctx context.Context, req admission.Request) admissio
 	}
 
-	for _, fcName := range fcNames {
-		ns, name := parseAnnotation(fcName, req.Namespace)
+	for _, fscName := range fscNames {
+		ns, name := parseAnnotation(fscName, req.Namespace)
 		if err != nil {
-			m.Log.V(1).Info(fmt.Sprintf("failed to parse annotation %s error: %s", fcName, err.Error()))
+			m.Log.V(1).Info(fmt.Sprintf("failed to parse annotation %s error: %s", fscName, err.Error()))
 			return admission.Errored(http.StatusBadRequest, err)
 		}
-		fc := m.getFlagSourceConfiguration(ctx, name, ns)
-		if reflect.DeepEqual(fc, corev1alpha1.FlagSourceConfiguration{}) {
-			m.Log.V(1).Info(fmt.Sprintf("FlagSourceConfiguration could not be found for %s", fcName))
+		fc := m.getFlagSourceConfiguration(ctx, ns, name)
+		if reflect.DeepEqual(fc, v1alpha1.FlagSourceConfiguration{}) {
+			m.Log.V(1).Info(fmt.Sprintf("FlagSourceConfiguration could not be found for %s", fscName))
 			return admission.Errored(http.StatusBadRequest, err)
 		}
 		flagSourceConfigurationSpec.Merge(&fc.Spec)
 	}
 
-	ffConfigs := []*corev1alpha1.FeatureFlagConfiguration{}
-	for _, ffName := range ffNames {
-		ns, name := parseAnnotation(ffName, req.Namespace)
-		if err != nil {
-			m.Log.V(1).Info(fmt.Sprintf("failed to parse annotation %s error: %s", ffName, err.Error()))
-			return admission.Errored(http.StatusBadRequest, err)
+	// maintain backwards compatibility of the openfeature.dev/featureflagconfiguration annotation
+	ffConfigAnnotation, ffConfigAnnotationOk := pod.GetAnnotations()["openfeature.dev/featureflagconfiguration"]
+	if ffConfigAnnotationOk {
+		m.Log.V(1).Info("DEPRECATED: The openfeature.dev/featureflagconfiguration annotation has been superseded by the openfeature.dev/flagsourceconfiguration annotation. " +
+			"Docs: https://github.com/open-feature/open-feature-operator/blob/main/docs/annotations.md")
+		if err := m.handleFeatureFlagConfigurationAnnotation(ctx, flagSourceConfigurationSpec, ffConfigAnnotation, req.Namespace); err != nil {
+			m.Log.Error(err, "unable to handle openfeature.dev/featureflagconfiguration annotation")
+			return admission.Errored(http.StatusInternalServerError, err)
 		}
-		// Check to see whether the FeatureFlagConfiguration has service or sync overrides
-		ff := m.getFeatureFlag(ctx, name, ns)
-		if reflect.DeepEqual(ff, corev1alpha1.FeatureFlagConfiguration{}) {
-			m.Log.V(1).Info(fmt.Sprintf("FeatureFlagConfiguration could not be found for %s", ffName))
-			return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	marshaledPod, err := m.injectSidecar(ctx, pod, flagSourceConfigurationSpec)
+	if err != nil {
+		m.Log.Error(err, "unable to inject flagd sidecar")
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+}
+
+func (m *PodMutator) handleFeatureFlagConfigurationAnnotation(ctx context.Context, fcConfig *v1alpha1.FlagSourceConfigurationSpec, ffconfigAnnotation string, defaultNamespace string) error {
```

Comment:
perhaps this func could be moved to a `pod_webhook_deprecated.go` file?

Reply:
great suggestion